source/sync/data-access-patterns/permissions.txt

```diff
@@ -360,6 +353,183 @@ change until the next session.
+      "write": { "teamId": "%%user.custom_data.teamId" },
+      "fields": {
+        "address": {
+          "fields": {
+            "zipCode": {
+              "write": false,
+              "read": true
+            }
+          },
+          "additional_fields": {
+            "write": true
+          }
+        },
+        "name": {}
+      },
+      "additional_fields": {
+        "write": false,
+        "read": false
+      }
+    }
+
+
+A **Role** object has the following properties:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 20 10 50
+
+   * - Key
+     - Value Type
+     - Required
+     - Notes
+
+   * - name
+     - string
+     - Yes
+     - The name of the role.
+
+   * - applyWhen
+     - JSON expression
+     - Yes
+     - The rules system evaluates the expression at session start time
+       and assigns the first role whose ``applyWhen`` expression evaluates to
+       true.
+
+   * - write
+     - JSON expression
+     - No
+     - The rules system evaluates the expression for each document to
+       determine if the user may write to the given document. If true, the
+       system implicitly grants read access. The expression may only refer to
+       the pre-specified **queryable fields** of the document.
+
+   * - read
+     - JSON expression
+     - No
+     - The rules system evaluates the expression for each document to
+       determine if the user may write to the given document. If ``write``
+       evaluates to true, the system implicitly grants read access regardless of
+       the result of ``read``. The expression may only refer to the
+       pre-specified **queryable fields** of the document.
+
+   * - fields
+     - Fields object
+     - No
+     - Specifies field-level permissions in addition to document-level
+       permission. If omitted, document-level rules apply.
+
+   * - additional_fields
+     - Object
+     - No
+     - Specifies permissions for fields not explicitly listed in the
+       ``fields`` object.
+
+   * - additional_fields.read
+     - boolean
+     - No
+     - Specifies whether read access should be granted for fields not
+       explicitly listed in the ``fields`` object.
+       
+   * - additional_fields.write
+     - boolean
+     - No
+     - Specifies whether write access should be granted for fields not
+       explicitly listed in the ``fields`` object.
+
+The **Fields** object defines permissions for specific fields. The keys of the
+fields object are names of the field:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 20 10 50
+
+   * - Key
+     - Value Type
+     - Required
+     - Notes
+
+   * - *arbitrary field name*
+     - **Permissions** object
+     - No
+     - Specifies the field-level permissions for the given field.
+
+The **Permissions** object has the following properties:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 20 10 50
+
+   * - Key
+     - Value Type
+     - Required
+     - Notes
+
+   * - write
```

Comment:
Do you think it would be useful to also say that undefined "write" and "read" by default mean that the field has access?

Reply:
I get how it works in that we are assumed to have access by the time we even look at the field-level permissions, but I'm concerned about how to communicate that practically. What if we say "if undefined, it grants the document-level permission for this field" or something like that?

Reply:
Yeah I think that's fine